controllers/account.go

```diff
@@ -176,8 +176,12 @@ func (c *ApiController) Signup() {
 
 	username := authForm.Username
 	if !application.IsSignupItemVisible("Username") {
-		if organization.UseEmailAsUsername && application.IsSignupItemVisible("Email") {
-			username = authForm.Email
+		if organization.UseEmailAsUsername {
+			if application.IsSignupItemVisible("Email") {
+				username = authForm.Email
+			} else {
+				username = util.GetUsernameFromEmail(authForm.Email)
```

Comment:
I think if Email is invisible in signup item, authForm.Email will be empty, so why pass any empty string to this function and get some random string? It will not generate a virtual email address .